src/iot/fieldCategories.js

```diff
@@ -0,0 +1,216 @@
+   *      organizationId: '8a3cb818-0889-4674-b871-7ceadaecd26a',
+   *      parentCategoryId: 'e9f8f89c-609c-4c83-8ebc-cea928af661e'
+   * })
+   *   .then((fieldCategory) => console.log(fieldCategory))
+   *   .catch((err) => console.log(err));
+   */
+  create(fieldCategory) {
+    const requiredFields = ['name', 'description', 'organizationId'];
+
+    for (let i = 0; requiredFields.length > i; i++) {
+      const field = requiredFields[i];
+      if (!fieldCategory[field]) {
+        return Promise.reject(
+          new Error(`A ${field} is required to create a new field category.`)
+        );
+      }
+    }
+
+    const formattedCategory = toSnakeCase(fieldCategory);
+
+    return this._request
+      .post(`${this._baseUrl}/categories`, formattedCategory)
+      .then((fieldCategory) => toCamelCase(fieldCategory));
+  }
+
+  /**
+   * Deletes a field category
+   *
+   * API Endpoint: '/categories/:categoryId'
+   * Method: DELETE
+   *
+   * @param {String} categoryId The UUID of a field category
+   *
+   * @returns {Promise}
+   * @fulfill {undefined}
+   * @reject {Error}
+   *
+   * @example
+   * contxtSdk.iot.fieldCategories
+   *   .delete('b3dbaae3-25dd-475b-80dc-66296630a8d0');
+   */
+  delete(categoryId) {
+    if (!categoryId) {
+      return Promise.reject(
+        new Error('A categoryId is required for deleting a field category.')
+      );
+    }
+
+    return this._request.delete(`${this._baseUrl}/categories/${categoryId}`);
+  }
+
+  /**
+   * Gets information about a field category
+   *
+   * API Endpoint: '/categories/:categoryId'
+   * Method: GET
+   *
+   * @param {String} categoryId The UUID of a field category
+   *
+   * @returns {Promise}
+   * @fulfill {FieldCategory} Information about the field category
+   * @reject {Error}
+   *
+   * @example
+   * contxtSdk.iot.fieldCategories
+   *   .get('b3dbaae3-25dd-475b-80dc-66296630a8d0')
+   *   .then((fieldCategory) => console.log(fieldCategory))
+   *   .catch((err) => console.log(err));
+   */
+  get(categoryId) {
+    if (!categoryId) {
+      return Promise.reject(
+        new Error(
+          'A categoryId is required for getting information about a field category.'
+        )
+      );
+    }
+
+    return this._request
+      .get(`${this._baseUrl}/categories/${categoryId}`)
+      .then((fieldCategory) => toCamelCase(fieldCategory));
+  }
+
+  /**
+   * Get a listing of all field categories available to the user.
+   *
+   * API Endpoint: '/categories'
+   * Method: GET
+   *
+   * @returns {Promise}
+   * @fulfill {FieldCategory[]} Information about the field categories
+   * @reject {Error}
+   *
+   * @example
+   * contxtSdk.iot.fieldCategories
+   *   .getAll()
+   *   .then((fieldCategories) => console.log(fieldCategories))
+   *   .catch((err) => console.log(err));
+   */
+  getAll() {
+    return this._request
+      .get(`${this._baseUrl}/categories`)
+      .then((fieldCategories) => toCamelCase(fieldCategories));
```

Comment:
I believe this endpoint is paginated. Should handle it differently and documentation should be different. Additionally, we've been making an effort to not name gets to paginated endpoints with `getAll` since it doesn't get all, but I'm not real sure what would be a better name in this case.